script.js

```diff
@@ -0,0 +1,886 @@
+                this.textarea.selectionStart += 1;
+                console.log(this.caretPos, this.caretPosEnd);
+                break;
+            case "ArrowLeft":
+                this.textarea.selectionStart -= 1;
+                this.textarea.selectionEnd -= 1;
+                console.log(this.caretPos, this.caretPosEnd);
+                break;
+            case "ArrowUp":
+                this.textarea.textContent += "▲";
+                console.log(this.caretPos, this.caretPosEnd);
+                break;
+            case "ArrowDown":
+                this.textarea.textContent += "▼";
+                break;
+        }
+    }
+
+
+    // ====== Enter move ======
+    enter_move(event) {
+        let mod_e;
+        if (event.type == 'click') {
+            mod_e = event.target.getAttribute('data-code');
+        } else {
+            mod_e = event.code;
+        }
+        if (mod_e == 'Enter') {
+            this.caret_move('\r\n');
+        }
+    }
+
+    // ====== Space move ======
+    space_move(event) {
+        let mod_e;
+        if (event.type == 'click') {
+            mod_e = event.target.getAttribute('data-code');
+        } else {
+            mod_e = event.code;
+        }
+        if (mod_e == 'Space') {
+            this.caret_move(' ');
+        }
+    }
+
+    // ====== Tab move ======
+    tab_move(event) {
+        let mod_e;
+        if (event.type == 'click') {
+            mod_e = event.target.getAttribute('data-code');
+        } else {
+            mod_e = event.code;
+        }
+        if (mod_e == 'Tab') {
+            this.caret_move('\t');
+        }
+    }
+
+    // ====== Shift move and add active class ======
+    keydown(event) {
+        event.preventDefault();
+        let mod_e;
+        if (event.type == 'mousedown') {
+            mod_e = event.target.getAttribute('data-code');
+        } else {
+            mod_e = event.code;
+        }
+        switch (mod_e) {
+            case 'CapsLock':
+                switch (this.caps) {
+                    case true:
+                        this.caps = false;
+                        this.language_case_keyboard();
+                        this.key_elements[mod_e].classList.remove('active_key');
+                        break;
+                    case false:
+                        this.caps = true;
+                        this.language_case_keyboard();
+                        this.key_elements[mod_e].classList.add('active_key');
+                        break;
+                }
+                break;
+            case 'ShiftLeft':
+                this.shift = true;
+                this.language_case_keyboard();
+                this.key_elements[mod_e].classList.add('active_key');
+                break;
+            case 'ShiftRight':
+                this.shift = true;
+                this.language_case_keyboard();
+                this.key_elements[mod_e].classList.add('active_key');
+                break;
```

Comment:
```suggestion
            case 'ShiftLeft':
            case 'ShiftRight':
                this.shift = true;
                this.language_case_keyboard();
                this.key_elements[mod_e].classList.add('active_key');
                break;
```

Will work absolutely same. Please read how switch-case works.